core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -439,6 +451,22 @@ default Builder retryPolicy(RetryMode retryMode) {
         Builder addMetricPublisher(MetricPublisher metricPublisher);
 
         List<MetricPublisher> metricPublishers();
+
+        /**
+         * Sets the additional execution attributes collection for this client.
+         * @param executionAttributes Execution attributes map for this client.
+         * @return This object for method chaining.
+         */
+        Builder executionAttributes(Map<ExecutionAttribute<?>, Object> executionAttributes);
+
+        /**
+         * Add an execution attribute to the existing collection of execution attributes.
+         * @param attribute The execution attribute object
+         * @param value The value of the execution attribute.
+         */
+        Builder addExecutionAttribute(ExecutionAttribute attribute, Object value);
```

Comment:
- Let's use the `put` verb here. Typically `add` is used if you're adding to a existing set under the given key.
 - Can we make this typed? i.e. `addExecutionAttribute(ExecutionAttribute<T> attribute, T value)`